scoring/score.py

```diff
@@ -45,6 +45,49 @@ def calculate_scores(self):
         }
 
 
+class EndStateScorer(object):
+    # Assumption: we will trust the *ordering* of the entries in the data we
+    # are given and *ignore* the actual 'time' values. We defer handling of
+    # equialent-time operations to the territory controller.
+
+    def __init__(self, teams_data, arena_data):
+        self._zone_to_tla = {
+            info['zone']: tla
+            for tla, info in teams_data.items()
+        }
+
+        # Mapping from station_code -> owning zone
+        self._final_state = {
+            claim['station_code']: claim['zone']
+            for claim in arena_data['other']['territory_claims']
+        }
+
+    def calculate_scores(self):
+        zone_to_territories = collections.defaultdict(list)
+        for territory in self._final_state:
+            zone_to_territories[self._final_state[territory]].append(territory)
```

Comment:
Suggest using `.items()` here to avoid a redundant dict lookup:
```suggestion
        for territory, zone in self._final_state.items():
            zone_to_territories[zone].append(territory)
```